data/labels.go

```diff
@@ -80,6 +81,33 @@ func (l Labels) String() string {
 	return sb.String()
 }
 
+type Fingerprint uint64
+
+// Fingerprint calculates a 64-bit FNV-1 hash of the labels. Labels are sorted by key to make sure the hash is stable.
+func (l Labels) Fingerprint() Fingerprint {
+	h := fnv.New64()
+	if len(l) == 0 {
+		return Fingerprint(h.Sum64())
+	}
+	// maps do not guarantee predictable sequence of keys.
+	// Therefore, to make hash stable, we need to sort keys
+	keys := make([]string, 0, len(l))
+	for labelName := range l {
+		keys = append(keys, labelName)
+	}
+	sort.Strings(keys)
+	for _, name := range keys {
+		// avoid an extra allocation of a slice of bytes.
+		// The internal structure of the string is almost like as slice (except capacity).
+		_, _ = h.Write(unsafe.Slice(unsafe.StringData(name), len(name)))
```

Comment:
what's the difference between this and `h.Write([]byte(name))`?

Reply:
also, can you handle errors? We don't have a lint rule (yet) but errors should be handled

Reply:
> what's the difference between this and h.Write([]byte(name))?

conversion to `[]byte` causes a copy of the string to a new slice. This happens to make sure that string is immutable. My implementation avoids the unnecessary allocation because it is used only for read operation. 

> also, can you handle errors? We don't have a lint rule (yet) but errors should be handled

fnv implements io.Writer interface but the implementation of the Write method never returns error https://github.com/golang/go/blob/master/src/hash/fnv/fnv.go#L119-L127. In the Grafana code base we do not usually handle errors in this case. I searched for usages of `hash.New` in other go projects and almost everywhere the error is ignored.

Reply:
okay, thanks for the explanation. Can you add a comment summarizing that? (so anyone looking at this code gets the reason)